// @aws-sdk/client-ssm/dist/es/commands/DescribeMaintenanceWindowExecutionTaskInvocationsCommand.js
import { __extends } from "tslib";
import { DescribeMaintenanceWindowExecutionTaskInvocationsRequest, DescribeMaintenanceWindowExecutionTaskInvocationsResult, } from "../models/models_0";
import { deserializeAws_json1_1DescribeMaintenanceWindowExecutionTaskInvocationsCommand, serializeAws_json1_1DescribeMaintenanceWindowExecutionTaskInvocationsCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Retrieves the individual task executions (one per target) for a particular task run as part
 *    of a maintenance window execution.</p>
 */
var DescribeMaintenanceWindowExecutionTaskInvocationsCommand = /** @class */ (function (_super) {
    __extends(DescribeMaintenanceWindowExecutionTaskInvocationsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeMaintenanceWindowExecutionTaskInvocationsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeMaintenanceWindowExecutionTaskInvocationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SSMClient";
        var commandName = "DescribeMaintenanceWindowExecutionTaskInvocationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeMaintenanceWindowExecutionTaskInvocationsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeMaintenanceWindowExecutionTaskInvocationsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeMaintenanceWindowExecutionTaskInvocationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeMaintenanceWindowExecutionTaskInvocationsCommand(input, context);
    };
    DescribeMaintenanceWindowExecutionTaskInvocationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeMaintenanceWindowExecutionTaskInvocationsCommand(output, context);
    };
    return DescribeMaintenanceWindowExecutionTaskInvocationsCommand;
}($Command));
export { DescribeMaintenanceWindowExecutionTaskInvocationsCommand };
