// @aws-sdk/client-ssm/dist/es/commands/DescribePatchGroupStateCommand.js
import { __extends } from "tslib";
import { DescribePatchGroupStateRequest } from "../models/models_0";
import { DescribePatchGroupStateResult } from "../models/models_1";
import { deserializeAws_json1_1DescribePatchGroupStateCommand, serializeAws_json1_1DescribePatchGroupStateCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Returns high-level aggregated patch compliance state for a patch group.</p>
 */
var DescribePatchGroupStateCommand = /** @class */ (function (_super) {
    __extends(DescribePatchGroupStateCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribePatchGroupStateCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribePatchGroupStateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SSMClient";
        var commandName = "DescribePatchGroupStateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribePatchGroupStateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribePatchGroupStateResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribePatchGroupStateCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribePatchGroupStateCommand(input, context);
    };
    DescribePatchGroupStateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribePatchGroupStateCommand(output, context);
    };
    return DescribePatchGroupStateCommand;
}($Command));
export { DescribePatchGroupStateCommand };
