// @aws-sdk/client-ssm/dist/es/commands/GetInventoryCommand.js
import { __extends } from "tslib";
import { GetInventoryRequest, GetInventoryResult } from "../models/models_1";
import { deserializeAws_json1_1GetInventoryCommand, serializeAws_json1_1GetInventoryCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Query inventory information.</p>
 */
var GetInventoryCommand = /** @class */ (function (_super) {
    __extends(GetInventoryCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function GetInventoryCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    GetInventoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SSMClient";
        var commandName = "GetInventoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetInventoryRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetInventoryResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetInventoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetInventoryCommand(input, context);
    };
    GetInventoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetInventoryCommand(output, context);
    };
    return GetInventoryCommand;
}($Command));
export { GetInventoryCommand };
