// @aws-sdk/client-ssm/dist/es/commands/GetMaintenanceWindowExecutionTaskInvocationCommand.js
import { __extends } from "tslib";
import { GetMaintenanceWindowExecutionTaskInvocationRequest, GetMaintenanceWindowExecutionTaskInvocationResult, } from "../models/models_1";
import { deserializeAws_json1_1GetMaintenanceWindowExecutionTaskInvocationCommand, serializeAws_json1_1GetMaintenanceWindowExecutionTaskInvocationCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Retrieves information about a specific task running on a specific target.</p>
 */
var GetMaintenanceWindowExecutionTaskInvocationCommand = /** @class */ (function (_super) {
    __extends(GetMaintenanceWindowExecutionTaskInvocationCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function GetMaintenanceWindowExecutionTaskInvocationCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    GetMaintenanceWindowExecutionTaskInvocationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SSMClient";
        var commandName = "GetMaintenanceWindowExecutionTaskInvocationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetMaintenanceWindowExecutionTaskInvocationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetMaintenanceWindowExecutionTaskInvocationResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetMaintenanceWindowExecutionTaskInvocationCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetMaintenanceWindowExecutionTaskInvocationCommand(input, context);
    };
    GetMaintenanceWindowExecutionTaskInvocationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetMaintenanceWindowExecutionTaskInvocationCommand(output, context);
    };
    return GetMaintenanceWindowExecutionTaskInvocationCommand;
}($Command));
export { GetMaintenanceWindowExecutionTaskInvocationCommand };
